include/boost/url/detail/remove_dot_segments.hpp

```diff
@@ -0,0 +1,51 @@
+//
+// Copyright (c) 2022 alandefreitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/CPPAlliance/url
+//
+
+#ifndef BOOST_URL_DETAIL_REMOVE_DOT_SEGMENTS_HPP
+#define BOOST_URL_DETAIL_REMOVE_DOT_SEGMENTS_HPP
+
+#include <boost/url/detail/config.hpp>
+#include <boost/url/string_view.hpp>
+
+namespace boost {
+namespace urls {
+namespace detail {
+
+BOOST_URL_DECL
+std::size_t
+remove_dot_segments(
+    char* dest,
+    char const* end,
+    string_view s,
+    bool remove_leading_dot_dot = true) noexcept;
+
+template<
+    class Allocator = std::allocator<char> >
+std::basic_string<char,
+    std::char_traits<char>,
+        Allocator>
+remove_dot_segments(
```

Comment:
Is this used?

Reply:
Not really. I used to be used. But it's just being used in tests now. I'll remove it and allocate the string directly in the tests.